src/coreclr/jit/optimizebools.cpp

```diff
@@ -1031,8 +1034,8 @@ bool OptBoolsDsc::optOptimizeCompareChainCondBlock()
         int op1Cost = cond1->GetCostEx();
         int op2Cost = cond2->GetCostEx();
         // The cost of combing three simple conditions is 32.
-        int maxOp1Cost = op1IsCondChain ? 31 : 7;
-        int maxOp2Cost = op2IsCondChain ? 31 : 7;
+        int maxOp1Cost = op1IsCondChain ? 31 : 10;
```

Comment:
What are these changes?

Reply:
When I originally added these costs, they were just best guesses to ensures we don't optimise for long chains. Tweaking it should be fine.